server/job_stream.go

```diff
@@ -0,0 +1,1293 @@
+			newToken, err := getToken(ctx, 3*time.Second, token.ServiceAddr, stream.Pipeline, jobSender.Addr, jobSender.Sig)
+			if err != nil {
+				clog.Errorf(ctx, "Error getting new token for %s: %v", token.ServiceAddr, err)
+				continue
+			}
+			newSess, err := tokenToAISession(*newToken)
+			if err != nil {
+				clog.Errorf(ctx, "Error converting token to AI session: %v", err)
+				continue
+			}
+			params.liveParams.sess = &newSess
+			stream.UpdateStreamParams(params)
+
+			// send the payment
+			jobDetails := JobRequestDetails{StreamId: streamId}
+			jobDetailsStr, err := json.Marshal(jobDetails)
+			if err != nil {
+				clog.Errorf(ctx, "Error marshalling job details: %v", err)
+				continue
+			}
+			req := &JobRequest{Request: string(jobDetailsStr), Parameters: "{}", Capability: stream.Pipeline,
+				Sender:  ls.LivepeerNode.OrchestratorPool.Broadcaster().Address().Hex(),
+				Timeout: 60,
+			}
+			//sign the request
+			job := gatewayJob{Job: &orchJob{Req: req}, node: ls.LivepeerNode}
+			err = job.sign()
+			if err != nil {
+				clog.Errorf(ctx, "Error signing job, continuing monitoring: %v", err)
+				continue
+			}
+
+			pmtHdr, err := createPayment(ctx, req, newToken, ls.LivepeerNode)
+			if err != nil {
+				clog.Errorf(ctx, "Error processing stream payment for %s: %v", streamId, err)
+				// Continue monitoring even if payment fails
+			}
+			if pmtHdr == "" {
+				continue
+			}
+
+			//send the payment, update the stream with the refreshed token
+			clog.Infof(ctx, "Sending stream payment for %s", streamId)
+			statusCode, err := ls.sendPayment(ctx, token.ServiceAddr+"/ai/stream/payment", stream.Pipeline, job.SignedJobReq, pmtHdr)
+			if err != nil {
+				clog.Errorf(ctx, "Error sending stream payment for %s: %v", streamId, err)
+				continue
+			}
+			if statusCode != http.StatusOK {
+				clog.Errorf(ctx, "Unexpected status code %d received for %s", statusCode, streamId)
+				continue
+			}
+		}
+	}
+}
+
+type StartRequest struct {
+	Stream     string `json:"stream_name"`
+	RtmpOutput string `json:"rtmp_output"`
+	StreamId   string `json:"stream_id"`
+	Params     string `json:"params"`
+}
+
+type StreamUrls struct {
+	StreamId      string `json:"stream_id"`
+	WhipUrl       string `json:"whip_url"`
+	WhepUrl       string `json:"whep_url"`
+	RtmpUrl       string `json:"rtmp_url"`
+	RtmpOutputUrl string `json:"rtmp_output_url"`
+	UpdateUrl     string `json:"update_url"`
+	StatusUrl     string `json:"status_url"`
+	DataUrl       string `json:"data_url"`
+}
+
+func (ls *LivepeerServer) setupStream(ctx context.Context, r *http.Request, job *gatewayJob) (*StreamUrls, int, error) {
+	if job == nil {
+		return nil, http.StatusBadRequest, errors.New("invalid job")
+	}
+
+	requestID := string(core.RandomManifestID())
+	ctx = clog.AddVal(ctx, "request_id", requestID)
+
+	// Setup request body to be able to preserve for retries
+	// Read the entire body first with 10MB limit
+	bodyBytes, err := io.ReadAll(io.LimitReader(r.Body, 10<<20))
```

Comment:
Since io.ReadALL truncates to 10MB here, maybe we want to log that?
```suggestion
if len(bodyBytes) > 10<<20 {
	fmt.Errorf("request body too large, maximum size is %d bytes", 10<<20)
}
```

Reply:
Updated how the r.Body is limited to 10MB so that can get a specific error in 67cb9cd8. Looked like from docs that limit reader would just stop reading and provide no error.